audit_engine/dynamic_analysis/__init__.py

```
@@ -100,52 +222,77 @@ async def analyze_contracts(
-            # Collect results with timeout handling
-            for adapter, future in futures:
+            # Collect results with proper error handling
+            for future in as_completed(future_to_adapter):
+                adapter = future_to_adapter[future]
                 try:
-                    adapter_results = future.result(timeout=self.config.get("analysis_timeout", 600))
+                    timeout = self.config.get("analysis_timeout", DEFAULT_ANALYSIS_TIMEOUT)
+                    adapter_results = future.result(timeout=timeout)
                     processed_results = self._process_adapter_results(adapter, adapter_results)
                     results.extend(processed_results)
                 except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)